ibis/expr/api.py

```diff
@@ -1031,4 +1032,40 @@ def get_backend(expr: Expr | None = None) -> BaseBackend:
 intersect = ir.Table.intersect
 difference = ir.Table.difference
 
-_ = Deferred()
+_ = deferred = Deferred()
+"""Deferred expression object.
+
+Use this object to refer to a previous table expression in chain of
+expressions.
+
+!!! note "`_` may conflict with other idioms in Python"
+
+    See https://github.com/ibis-project/ibis/issues/4704 for details.
+
+    Use `from ibis import deferred as <NAME>` to assign a different name to
+    the deferred object builder.
+
+Examples
+--------
+>>> from ibis import _
+>>> t = ibis.table(dict(key="int", value="float"), name="t")
+>>> t.group_by(key=_.key - 1).agg(total=_.value.sum())
+r0 := UnboundTable: t
+  key   int64
+  value float64
+<BLANKLINE>
+Aggregation[r0]
+  metrics:
+    total: Sum(r0.value)
+  by:
+    key: r0.key - 1
+>>> t.mutate(other_value=_.value * 2)
+r0 := UnboundTable: t
+  key   int64
+  value float64
+<BLANKLINE>
+Selection[r0]
+  selections:
+    r0
+    other_value: r0.value * 2
```

Comment:
Since our repr sometimes changes (and IMO doesn't provide the user with a ton of actionable info), I've taken to removing the expr reprs from docstrings. You might assign the results of these expressions to a variable instead:

```python
expr2 = t.mutate(other_value=_.value * 2)
```

Reply:
Fair, I can remove it. I always find the assignment leaves me wanting for some kind of resolution though.

Reply:
I'll show the schema repr, that hasn't changed in forever.

Reply:
Done.